build-packages/scandipwa-development-toolkit-core/src/util/generate-files-from-map.ts

```diff
@@ -0,0 +1,80 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { FileMap, ILogger, ResourceType } from '../types';
+import { createNewFileFromTemplate } from './file';
+
+const templateDirectory = path.resolve(__dirname, '..', 'templates');
+
+const getIsNested = (resourceType: ResourceType) => resourceType !== ResourceType.Query;
+
+/**
+ * Generate files from the filemap
+ * Returns relative paths to the created files
+ * 
+ * @param fileMap 
+ * @param options 
+ * @returns {string[]}
+ */
+const generateFilesFromMap = (
+    fileMap: FileMap, 
+    resourceName: string, 
+    resourceType: ResourceType, 
+    targetModulePath: string,
+    logger: ILogger
+): string[] => {
+    // Calculate the resource path
+    const resourceTypeDirectory = path.join(targetModulePath, 'src', resourceType);
+    const resourceDirectory = getIsNested(resourceType)
+        ? path.join(resourceTypeDirectory, resourceName)
+        : resourceTypeDirectory;
+
+    // Ensure parent direcotory exists for files which'll be generated below
+    fs.ensureDirSync(resourceDirectory);
+
+    // Process entry point
+    const createdFiles: string[] = Object.entries(fileMap).reduce(
+        (createdFiles, [postfix, templateName]) => {
+            // Handle skippable entries
+            if (!templateName || !postfix) {
+                return createdFiles;
+            }
+
+            // Calculate the template path
+            const templatePath = path.join(templateDirectory, templateName);
+
+            // Index.js is not a postfix, it should be handled differently
+            const newFileName = postfix === 'index.js' && 'index.js' 
```

Comment:
This constant `'index.js'` is always true :)

Reply:
This is not about booleans: if postfix is `index.js` here, the value will be `index.js` due to the `&&` operator.
Otherwise, the declaration below will be taken.
I will transform this to a ternary expression, so that it is more obvious :)